src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -508,9 +508,9 @@ internal void HandleWriteErrors(IndexingStatsScope stats, IndexWriteException iw
             if (iwe.InnerException is SystemException) // Don't count transient errors
                 return;
 
-            writeErrors = Interlocked.Increment(ref writeErrors);
+            _writeErrors = Interlocked.Increment(ref _writeErrors);
```

Comment:
Wrong you just incremented the value, and another thread incremented this as well, but then you go ahead and overwrite the value with an old value.